src/main/scala/workshop/p1/pure/examples/PureExample.scala

```diff
@@ -0,0 +1,33 @@
+package workshop.p1.pure.examples
+
+import org.scalatest.MustMatchers.convertToEqualizer
+
+
+object PureExample extends App {
+
+  // side effects free
+  def pureFunction(name: String): String = s"My name is $name"
+
+  // side effects
+  def impureFunction(name: String): Unit = println(s"My name is $name")
+
+
+  /// TEST PURE:
+  val resultOne = pureFunction("Andrzej")
+  pureFunction("Maciek")
+  pureFunction("Adam")
+
+  val expectedResult = "My name is Andrzej"
+
+  // result assert
+  assert(resultOne === expectedResult)
+  // Referential transparency assert
+  assert(expectedResult === pureFunction("Andrzej"))
+
+  // TEST IMPURE:
+  val resultImpure: Unit = impureFunction("Andrzej")
+  // assert(resultImpure === ?????)
+  // HOW ???
+  // get somehow access to output buffer ?????!!!
```

Comment:
I've been doing the same sort of workshop, if you need some inspiration you can see how I actually showed how hard is to do it imperatively here
https://github.com/CodeGardenRome/esercizi-funzionali-graduali/blob/scala/src/test/scala/Step00Spec.scala#L66

Reply:
thanks for example 👍 
I cannot include it for now because of workshop time constrains